frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/delete-block-pool-modal.tsx

```diff
@@ -0,0 +1,160 @@
+import { cephClusterResource } from '../../../resources';
+import {
+  blockPoolReducer,
+  blockPoolInitialState,
+  BlockPoolActionType,
+  ActionButtonTextMapping,
+} from '../../../utils/block-pool';
+import { commaSeparatedString } from '../../../utils/common';
+import { POOL_PROGRESS } from '../../../constants/storage-pool-const';
+import { CephBlockPoolModel } from '../../../models';
+import { CEPH_EXTERNAL_CR_NAME, OCS_INTERNAL_CR_NAME } from '../../../constants';
+import { getStorageClassName } from '../../../selectors';
+
+const DeleteBlockPoolModal = withHandlePromise((props: DeleteBlockPoolModalProps) => {
+  const { t } = useTranslation();
+  const { blockPoolConfig, cancel, close, handlePromise, inProgress } = props;
+  const poolName = blockPoolConfig?.metadata.name;
+
+  const [state, dispatch] = React.useReducer(blockPoolReducer, blockPoolInitialState);
+  const [scNames, setScNames] = React.useState<string>();
+
+  const [cephClusters, isLoaded, loadError] = useK8sWatchResource<CephClusterKind[]>(
+    cephClusterResource,
+  );
+  const [scResources, scLoaded, scLoadError] = useK8sGet<ListKind<OcsStorageClassResource>>(
+    StorageClassModel,
+  );
+  const [pvcResources, pvcLoaded, pvcLoadError] = useK8sGet<ListKind<PersistentVolumeClaimKind>>(
+    PersistentVolumeClaimModel,
+  );
+  const cephCluster: CephClusterKind = useDeepCompareMemoize(cephClusters[0], true);
+
+  React.useEffect(() => {
+    // restrict pool management for default pool and external cluster
+    cephCluster?.metadata.name === CEPH_EXTERNAL_CR_NAME ||
+    blockPoolConfig?.metadata?.ownerReferences?.find(
+      (ownerReference) => ownerReference.name === OCS_INTERNAL_CR_NAME,
+    )
+      ? dispatch({ type: BlockPoolActionType.SET_POOL_STATUS, payload: POOL_PROGRESS.NOTALLOWED })
+      : dispatch({
+          type: BlockPoolActionType.SET_POOL_NAME,
+          payload: poolName,
+        });
+  }, [blockPoolConfig, cephCluster, isLoaded, loadError, poolName]);
+
+  React.useEffect(() => {
+    if (scLoaded && pvcLoaded && state.poolStatus !== POOL_PROGRESS.NOTALLOWED) {
+      const poolScNames: string[] = scResources.items?.reduce((scList, sc) => {
+        if (sc.parameters?.pool === poolName) scList.push(sc.metadata?.name);
+        return scList;
+      }, []);
+      const pvcScNames: string[] = pvcResources.items?.map(getStorageClassName);
+
+      // intersection of scNames and pvcScNames
+      const usedScNames = poolScNames.filter((scName) => pvcScNames.includes(scName));
+
+      if (usedScNames.length) {
+        dispatch({ type: BlockPoolActionType.SET_POOL_STATUS, payload: POOL_PROGRESS.BOUNDED });
+        setScNames(commaSeparatedString(usedScNames, t));
+      }
+    }
+  }, [scResources, scLoaded, pvcResources, pvcLoaded, state.poolStatus, poolName, t]);
+
+  // Delete block pool
+  const onClick = () => {
+    handlePromise(k8sKill(CephBlockPoolModel, blockPoolConfig), () => close());
+  };
+
+  const redirectToPvcList = () => {
```

Comment:
i tried to keep this function inside the footer but it keeps crashing, so I decided to keep here